### ISSUE TITLE
Host on Whitelist is being blocked by gravity (v4.3.2)

### ISSUE THREAD

**Issue by maciejmiklas** [pi-hole/pi-hole#3265]
**Expected behaviour:**

Putting host on whitelist in Admin Console (HTTP client) should not block it anyone. 

**Actual behaviour:**
Host is on Whitelist and I can see in query log that it's being blocked. 
Action: Witelist (green)
Status: Blocked (gravity)

**Steps to reproduce:**
1) Admin Console -> Query Log 
2) Select blocked host and click Whitelist
3) Try to connect to this host in firefox, page is not found
4) Query Log  shows that host is on Whitelist and it's being blocked by Gravity

**Comment by PromoFaux** [comment]
Hi,

What specific host are you trying to whitelist? Could it be that there are other domains being queried that are preventing the page from loading? As far as I know there are no known bugs with whitelisting. 

Could you also please provide a debug token via `pihole -d`, thanks.

Another thing to check is your local cache on firefox.

**Comment by maciejmiklas** [comment]
Hi,

I've uploaded two screenshots: https://drive.google.com/open?id=186r7IyR-mMaMRnAod-N7QUVSV9zkykTq

You can see there, that collector.githubapp.com is whitelisted and blocked at the same time. I'm getting such log on every access to https://collector.githubapp.com/

**Comment by dschaper** [comment]
What does a `dig collector.githubapp.com @192.168.8.24` show as the IP?

Does that domain show with a `pihole -q -exact collector.githubapp.com` ?

**Comment by dschaper** [comment]
Also, it looks like that domain was not whitelisted. It's not in `/etc/pihole/whitelist.txt`

**Comment by maciejmiklas** [comment]
Yes - host was not on whitelist, but Query Log indicates otherwise.

**Comment by dschaper** [comment]
The Query Log showed that domain with the Status field "OK (forwarded)"?

**Comment by maciejmiklas** [comment]
The UI (see screenshots: https://drive.google.com/open?id=186r7IyR-mMaMRnAod-N7QUVSV9zkykTq) shows that domains are on whitelist. In my opinion it's a bug.

**Comment by dschaper** [comment]
The green check box in the Action field is not showing the domain is on the whitelist, it's a button to click to put the domain on the whitelist. Note the domains that are allowed have a red Blacklist button, that doesn't mean they are blacklisted.

**Comment by maciejmiklas** [comment]
Now I understand it! For some reason I was unable to add domains to whitelist from `Query Log`, but `Whitelist` menu works fine. 

When I was clicking on `Whitelist` button `Query Log` there was no confirmation that domain was included in whitelist (I also did not expect one). Now I've updated to latest version and I'm getting confirmation and domain is actually in whitelist.  So everything is fine - thank you!
